src/app/components/ATIAnalytics/params/index.ts

```diff
@@ -204,27 +212,48 @@ const createBuilderFactory = (
 };
 
 export const buildATIUrl = (
-  data: PageData,
   requestContext: RequestContextProps,
   serviceContext: ServiceConfig,
+  data?: PageData,
+  atiData?: ATIData,
 ) => {
-  const buildUrl = createBuilderFactory(requestContext, pageTypeUrlBuilders);
+  if (atiData) {
+    return buildPageATIUrl({ atiData, requestContext, serviceContext });
+  }
 
-  return buildUrl(data, requestContext, serviceContext);
+  if (data) {
+    return createBuilderFactory(requestContext, pageTypeUrlBuilders)(
+      data,
+      requestContext,
+      serviceContext,
+    );
+  }
+
+  return null;
 };
 
 export const buildATIEventTrackingParams = (
-  data: PageData,
+  data: PageData | HomePageProps,
   requestContext: RequestContextProps,
   serviceContext: ServiceConfig,
 ) => {
   try {
+    const atiData = pathOr(null, ['pageData', 'metadata'], data);
```

Comment:
We are trying to avoid having to do this - i.e. pass in entire page data objects. Can we follow a similar pattern to the buildATIUrl above, passing in ATIData as a prop